Commands/Provisioning/AddDataRowsToProvisioningTemplate.cs

```diff
@@ -0,0 +1,184 @@
+        public ITemplateProviderExtension[] TemplateProviderExtensions;
+
+    
+
+        protected override void ExecuteCmdlet()
+        {
+
+            
+            var template = LoadProvisioningTemplate
+                    .LoadProvisioningTemplateFromFile(Path,
+                    SessionState.Path.CurrentFileSystemLocation.Path,
+                    TemplateProviderExtensions);
+            
+            if (template == null)
+            {
+                throw new ApplicationException("Invalid template file!");
+            }
+    //We will remove a list if it's found so we can get the list
+
+            ListInstance listInstance = template.Lists.Find(l => l.Title == List.Title);
+            if (listInstance == null)
+            {
+                throw new ApplicationException("List does not exist in the template file!");
+            }
+            
+            List spList = List.GetList(SelectedWeb);
+            ClientContext.Load(spList, l => l.RootFolder, l => l.HasUniqueRoleAssignments);
+            ClientContext.ExecuteQueryRetry();
+
+            CamlQuery query = new CamlQuery();
+            var viewFieldsStringBuilder = new StringBuilder();
+
+            viewFieldsStringBuilder.Append("<ViewFields>");
+            foreach (var field in Fields)
+            {
+                viewFieldsStringBuilder.AppendFormat("<FieldRef Name='{0}'/>", field);
+            }
+            viewFieldsStringBuilder.Append("</ViewFields>");
+            query.ViewXml = string.Format("<View>{0}{1}</View>", Query, viewFieldsStringBuilder);
+            var listItems = spList.GetItems(query);
+            ClientContext.Load(listItems, lI=>lI.Include(l=>l.HasUniqueRoleAssignments, l=>l.ContentType.StringId));
+            ClientContext.ExecuteQueryRetry();
+
+            Microsoft.SharePoint.Client.FieldCollection fieldCollection = spList.Fields;
+            ClientContext.Load(fieldCollection, fs => fs.Include(f => f.InternalName, f => f.FieldTypeKind));
+            ClientContext.ExecuteQueryRetry();
+
+
+            var rows = new DataRowCollection(template);
+            foreach(var listItem in listItems)
+            {
+                //Make sure we don't pull Folders.. Of course this won't work
+                if (listItem.ServerObjectIsNull == false) {
+
+                    ClientContext.Load(listItem);
+                    ClientContext.ExecuteQueryRetry();
+                    if (!(listItem.FileSystemObjectType == FileSystemObjectType.Folder))
+                    {
+
+
+
+                        DataRow row = new DataRow();
+                        if (IncludeSecurity && listItem.HasUniqueRoleAssignments)
+                        {
+                            row.Security.ClearSubscopes = true;
+                            row.Security.CopyRoleAssignments = false;
+
+                            var RoleAssignments = listItem.RoleAssignments;
```

Comment:
casing standards for variables advices the variable to start with a lowercase char, unless it's a Property on the cmdlet